libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -99,6 +103,73 @@ string ABIFunctions::tupleEncoder(
 	});
 }
 
+string ABIFunctions::tupleDecoder(TypePointers const& _types, bool _fromMemory)
+{
+	string functionName = string("abi_decode_tuple_");
+	for (auto const& t: _types)
+		functionName += t->identifier();
+	if (_fromMemory)
+		functionName += "_fromMemory";
```

Comment:
Hm, wondering if we should commit to camelcase or underscores, this one mixes the two.